lib/service/awsoidc.go

```diff
@@ -335,3 +334,63 @@ func (updater *AWSOIDCDeployServiceUpdater) updateAWSOIDCDeployService(ctx conte
 	}
 	return nil
 }
+
+// getRegionsForUpdate is a helper that returns AWS regions to check for AWS
+// OIDC deploy services to update. In order to reduce the number of api
+// calls, the aws regions are reduced to only the regions containing an RDS
+// database or database service labeled as an integration agent.
+func getRegionsForUpdate(ctx context.Context, authClt authclient.ClientI) (iter.Seq[string], error) {
```

Comment:
Another thing we can check for is the DiscoveryConfig.
If it has an AWS Matcher which tries to find RDS databases and has an associated integration, then we should also collect the regions.

So, this ends up being very similar to what we have here
https://github.com/gravitational/teleport/pull/49352/files#diff-6361b250b7351118a68d7b41fbeb1d25fbdb7d88c030aff7fb265527c3a6496cR300

This PR is meant to get all teleport-managed ECS Services and return them to the user.
I think we should use the same implementation for getting the regions for both scenarios.

Reply:
I'm glad I requested your review! Yes we should either copy/paste your impl or put it in a common lib somewhere to call from both places.